### DISCUSSION TITLE
find files doesn't work when the file name contains some special characters

### DISCUSSION THREAD

**Discussion by varunbpatil** [LazyVim/LazyVim#5262] [Q&A] [answered]
Using fzf-lua.

1. Create a file named `foo::bar`.
2. `<leader>ff` to open find files.
3. search for the above file

The error I get is "Cannot stat file foo". This used to work earlier with telescope.nvim.

**Comment by dpetka2001** [comment]
Ask in fzf-lua repo? It uses `ripgrep` or `fd` behind the scenes, so it must have something to do with how those search engines interpret some characters. Unfortunately, I'm not that much in depth familiar with either of these 2 tools, so hopefully fzf-lua maintainer will be able to shed some light.

**Reply by dpetka2001** [reply]
It seems that the `:` character is being used internally by fzf-lua as delimiter, my guess is [this function](https://github.com/ibhagwan/fzf-lua/blob/dbc5bd14e72ba680df44906e957b0a821121726b/lua/fzf-lua/previewer/fzf.lua#L50-L66) for the previewer is what causes the error `Cannot stat file foo` to occur from quickly searching through the codebase and it seems that function, even if you try to set a custom delimiter in `opts.fzf_opts["--delimiter"]`, it still appends the character `:` at the end because it expects the entry to be of format `file:line:col: text`. 

I don't know if you can fully customize this behavior, so your best bet is the fzf-lua maintainer himself for more detailed insight.

**Reply by varunbpatil** [reply]
Thanks. Initiated discussion in fzf-lua.

**Comment by ibhagwan** [comment, marked as answer]
Fixed in https://github.com/ibhagwan/fzf-lua/commit/d99996b5838a30f4725e584bac5e8a0194e5a2a6

**Reply by varunbpatil** [reply]
Thanks. Can confirm that it works.

**Reply by ibhagwan** [reply]
> Thanks. Can confirm that it works.

Ty for the update @varunbpatil
